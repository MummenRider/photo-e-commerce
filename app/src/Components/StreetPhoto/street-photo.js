import {
  Container,
  SubTitle,
  Title,
  TextContainer,
  ImageBlockOne,
  ImageBlockTwo,
  ImageBlockThree,
  ImageBlockFour,
} from "Components/StreetPhoto/street-photo.styles";

export default function StreetPhoto({ isBigDevice, children, ...restProps }) {
  return isBigDevice ? (
    <Container data-testid="str-photo-component" {...restProps}>
      {children}
    </Container>
  ) : null;
}
StreetPhoto.TextContainer = function StreetPhotoTextContainer({
  children,
  ...restProps
}) {
  return <TextContainer {...restProps}>{children}</TextContainer>;
};
StreetPhoto.Title = function StreetPhotoTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

StreetPhoto.SubTitle = function StreetPhotoSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

StreetPhoto.ImageBlockOne = function StreetPhotoImageBlockOne({
  children,
  ...restProps
}) {
  return <ImageBlockOne {...restProps}>{children}</ImageBlockOne>;
};

StreetPhoto.ImageBlockTwo = function StreetPhotoImageBlockTwo({
  children,
  ...restProps
}) {
  return <ImageBlockTwo {...restProps}>{children}</ImageBlockTwo>;
};

StreetPhoto.ImageBlockThree = function StreetPhotoImageBlockThree({
  children,
  ...restProps
}) {
  return <ImageBlockThree {...restProps}>{children}</ImageBlockThree>;
};

StreetPhoto.ImageBlockFour = function StreetPhotoImageBlockFour({
  children,
  ...restProps
}) {
  return <ImageBlockFour {...restProps}>{children}</ImageBlockFour>;
};
