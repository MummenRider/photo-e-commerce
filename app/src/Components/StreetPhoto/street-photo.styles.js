import styled from "styled-components";

export const Container = styled.section`
  width: 100%;

  position: relative;
  height: 160vh;
  min-height: 1500px;
  margin: 300px auto;

  @media screen and (max-width: 1040px) {
    margin: 0;
  }
`;

export const TextContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: center;
  color: white;
  max-width: 60rem;
  min-width: 30rem;
  padding: 5rem 0;
  z-index: 5;
`;

export const Title = styled.h1`
  font-family: var(--ff-secondary);
  font-size: clamp(2.5rem, 4vw, 5rem);
  font-weight: 200;
  padding: 4rem 0;
  line-height: 1.5;
`;

export const SubTitle = styled.span`
  font-family: var(--ff-primary);
  font-weight: lighter;
  font-size: clamp(1.3rem, 0.93vw, 2.5rem);
  letter-spacing: 2px;
`;

export const ImageBlockOne = styled.div`
  position: absolute;
  height: 24%;
  width: fit-content;
  top: 200px;
  left: 150px;

  @media screen and (max-width: 1375px) {
    height: 22%;
  }

  @media screen and (max-width: 1230px) {
    height: 20%;
  }

  @media screen and (max-width: 1130px) {
    height: 18%;
  }
`;

export const ImageBlockTwo = styled.div`
  position: absolute;
  height: 18%;
  width: fit-content;
  top: 50px;
  right: 100px;

  @media screen and (max-width: 1375px) {
    height: 16%;
  }

  @media screen and (max-width: 1230px) {
    height: 14%;
  }

  @media screen and (max-width: 1130px) {
    height: 12%;
  }
`;
export const ImageBlockThree = styled.div`
  position: absolute;
  width: fit-content;
  height: 18%;
  bottom: 200px;
  left: 100px;

  @media screen and (max-width: 1375px) {
    height: 16%;
  }

  @media screen and (max-width: 1230px) {
    height: 14%;
    bottom: 300px;
  }

  @media screen and (max-width: 1130px) {
    height: 12%;
    bottom: 400px;
  }
`;
export const ImageBlockFour = styled.div`
  position: absolute;
  width: fit-content;
  height: 24%;
  bottom: 300px;
  right: 100px;

  @media screen and (max-width: 1375px) {
    height: 22%;
  }

  @media screen and (max-width: 1230px) {
    height: 20%;
    bottom: 420px;
  }

  @media screen and (max-width: 1130px) {
    height: 18%;
  }
`;
