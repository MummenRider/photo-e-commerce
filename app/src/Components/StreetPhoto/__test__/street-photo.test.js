import React from "react";
import { render } from "@testing-library/react";
import StreetPhoto from "Components/StreetPhoto/street-photo";

describe("<StreetPhoto />", () => {
  it("should render <StreetPhoto /> with populated data", () => {
    const { container, getByText } = render(
      <StreetPhoto isBigDevice={true}>
        <StreetPhoto.TextContainer>
          <StreetPhoto.SubTitle>COMMUNITY, LOVE, RESPECT</StreetPhoto.SubTitle>
          <StreetPhoto.Title>
            Lorem Ipsum Dolor Amet, Consetetur
          </StreetPhoto.Title>
          <StreetPhoto.SubTitle>Food Charity - Sikh</StreetPhoto.SubTitle>
        </StreetPhoto.TextContainer>

        {/* Image blocks is simply a container and therefore should be able to handle any kind of children passed to it */}
        <StreetPhoto.ImageBlockOne>Image content one</StreetPhoto.ImageBlockOne>
        <StreetPhoto.ImageBlockTwo>Image content two</StreetPhoto.ImageBlockTwo>
        <StreetPhoto.ImageBlockThree>
          Image content three
        </StreetPhoto.ImageBlockThree>
        <StreetPhoto.ImageBlockFour>
          Image content four
        </StreetPhoto.ImageBlockFour>
      </StreetPhoto>
    );

    expect(getByText("COMMUNITY, LOVE, RESPECT")).toBeTruthy();
    expect(getByText("Lorem Ipsum Dolor Amet, Consetetur")).toBeTruthy();
    expect(getByText("Food Charity - Sikh")).toBeTruthy();
    expect(getByText("Image content one")).toBeTruthy();
    expect(getByText("Image content two")).toBeTruthy();
    expect(getByText("Image content three")).toBeTruthy();
    expect(getByText("Image content four")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should not render <StreetPhoto /> in smaller devices", () => {
    const { container, queryByTestId } = render(
      <StreetPhoto isBigDevice={false}>
        <StreetPhoto.SubTitle>COMMUNITY, LOVE, RESPECT</StreetPhoto.SubTitle>
      </StreetPhoto>
    );

    expect(queryByTestId("str-photo-component")).toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });
});
