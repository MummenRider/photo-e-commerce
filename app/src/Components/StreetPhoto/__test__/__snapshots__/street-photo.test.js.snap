// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StreetPhoto /> should not render <StreetPhoto /> in smaller devices 1`] = `null`;

exports[`<StreetPhoto /> should render <StreetPhoto /> with populated data 1`] = `
<section
  class="sc-bdnxRM khCAqU"
  data-testid="str-photo-component"
>
  <div
    class="sc-gtsrHT cNKTQo"
  >
    <span
      class="sc-hKFxyN kFhYQn"
    >
      COMMUNITY, LOVE, RESPECT
    </span>
    <h1
      class="sc-dlnjwi fMROt"
    >
      Lorem Ipsum Dolor Amet, Consetetur
    </h1>
    <span
      class="sc-hKFxyN kFhYQn"
    >
      Food Charity - Sikh
    </span>
  </div>
  <div
    class="sc-eCApnc cdA-Das"
  >
    Image content one
  </div>
  <div
    class="sc-jSFjdj kqvPWF"
  >
    Image content two
  </div>
  <div
    class="sc-gKAaRy gKUFHy"
  >
    Image content three
  </div>
  <div
    class="sc-iCoGMd blfSAQ"
  >
    Image content four
  </div>
</section>
`;
